README.md

```diff
@@ -1,69 +1,77 @@
-# Dependency Injection Checker
+# Guice Checker
 
-A common problem when programming is TODO.
-This results in a run-time exception.
-
-The Dependency Injection Checker guarantees, at compile time, that your code will
-not suffer that run-time exception.
+One common challenge is dealing with dependencies and ensuring they are properly defined. Omitting or misconfiguring dependencies can lead to runtime errors, impacting the reliability and stability of code.
 
+The Guice Checker is a static analysis tool designed to address this issue by providing compile-time guarantees that your Guice dependency mappings are accurately defined.
 
 ## How to run the checker
 
-First, publish the checker to your local Maven repository by running
-`./gradlew publishToMavenLocal` in this repository.
+First, build the checker by running `./gradlew clean build` and then publish the checker to your local Maven repository by running `./gradlew publishToMavenLocal` in this repository.
 
 Then, if you use Gradle, add the following to the `build.gradle` file in
 the project you wish to type-check (using Maven is similar):
```

Comment:
I know that you didn't write this bit, but I'd suggest cutting "using Maven is similar", because it is not.